debezium-core/src/main/java/io/debezium/relational/mapping/MaskStrings.java

```diff
@@ -72,4 +101,56 @@ public Object convert(Object value) {
             return maskValue;
         }
     }
-}
+
+    @Immutable
+    protected static final class HashValueConverter implements ValueConverter {
+
+        private static final Logger logger = LoggerFactory.getLogger(HashValueConverter.class);
+        private final byte[] salt;
+        private final MessageDigest hashAlgorithm;
+
+        public HashValueConverter(byte[] salt, String hashAlgorithm) {
+            this.salt = salt;
+            try {
+                this.hashAlgorithm = MessageDigest.getInstance(hashAlgorithm);
+            }
+            catch (NoSuchAlgorithmException e) {
+                throw new IllegalArgumentException(e);
+            }
+        }
+
+        @Override
+        public Object convert(Object value) {
+            if (value instanceof Serializable) {
+                try {
+                    return toHash((Serializable) value);
+                }
+                catch (IOException e) {
+                    if (logger.isErrorEnabled()) {
+                        logger.error("can't calculate hash", e);
```

Comment:
Hum, hum. Perhaps better to mask it then? We'd loose the referential integrity, but probably still better than leaking unmasked values.

Reply:
returning null is probably better.